pkg/controller/syncsetinstance/syncsetinstance_controller_test.go

```diff
@@ -1503,3 +1503,40 @@ func computeHash(data interface{}) string {
 	}
 	return fmt.Sprintf("%x", md5.Sum(b))
 }
+
+func TestSSIReApplyDuration(t *testing.T) {
+	startProbe := time.Now()
+
+	tenMinutesAgo := time.Unix(metav1.NewTime(time.Now().Add(-10*time.Minute)).Unix(), 0)
+	fortyFiveMinutesAgo := time.Unix(metav1.NewTime(time.Now().Add(-45*time.Minute)).Unix(), 0)
+	ninetyMinutesAgo := time.Unix(metav1.NewTime(time.Now().Add(-90*time.Minute)).Unix(), 0)
+
+	resources := []runtime.Object{
+		testCM("foo1", "bar", "baz"),
+		testCM("foo2", "bar", "baz"),
+		testCM("foo3", "bar", "baz"),
+	}
+
+	cd := testClusterDeployment()
+	ss := testSyncSetWithResources("aaa", resources...)
+	ssi := syncSetInstanceForSyncSet(cd, ss)
+	ssi.Status = successfulResourceStatus(resources...)
+
+	oldestResourceApplyTimes := []time.Time{
+		tenMinutesAgo,
+		fortyFiveMinutesAgo,
+		ninetyMinutesAgo,
+	}
+
+	for _, oldestResourceApplyTime := range oldestResourceApplyTimes {
+		ssi.Status.Resources[0].Conditions[0].LastProbeTime = metav1.NewTime(oldestResourceApplyTime)
```

Comment:
This test is only setting the time in the first resource. We do not need to test that multiple different times in the first resource works. We need to test that the right time is picked regardless of which resource has the latest time and regardless of where in the conditions slice the apply-success condition is.

Reply:
Modified this to change the time for each resource in the loop since I had 3 example times.